internal/controller/storageclient_controller.go

```diff
@@ -779,27 +661,33 @@ func removeStorageClaimAsOwner(obj client.Object) {
 	}
 }
 
-func (r *storageClientReconcile) EnsureClusterScopedResource(obj client.Object, extRes *providerpb.ExternalResource, useReplace bool) error {
-	obj.SetName(extRes.Name)
+func (r *storageClientReconcile) ensureResource(obj client.Object, rawObject []byte, useReplace bool, clusterScoped bool) error {
+	objectMeta := metav1.ObjectMeta{}
+	if err := json.Unmarshal(rawObject, &objectMeta); err != nil {
+		return err
+	}
+
+	obj.SetName(objectMeta.Name)
```

Comment:
What is the point in setting the name if the unmarsheling will override it?

Reply:
We need to name in order to createOrUpdate/createOrReplcace the object